### ISSUE TITLE
Why does guake has ists own tabs implementation

### ISSUE THREAD

**Issue by aichingm** [Guake/guake#1375]
Hey @gsemet,

is there a reason why guake does not use the Gtk.Notebook's tab implementation but a custom radio button group?

cheers Mario

**Comment by gsemet** [comment]
Hi. Not sure why, probably a very old code. I am about to rewrite this part to enable much expected feature, but this part is kind of ugly I need to break everything :(

**Comment by aichingm** [comment]
Are you working on it right now? I just started to work on split terminal functionality, and rewriting the notebook part.

**Comment by aichingm** [comment]
The things I thought about doing are: 
1. remove the custom tabs and use the notebook's
1. resolve `# TODO PORT ...` sections (finally)
1. rewrite the notebook to allow multiple terminals on one page and remove the term_list 
1. also move the functions rename_tab, delete_tab and add_tab to the notebook
1. implement split terminal using Gtk.Paned (I know this is not as flexible as the split functionality of terminator but I see no point in reimplementing terminator, that would be a hell of a task) and a bunch of recursive calls on the widget hierarchy . 

Do you have any ideas, opinions or inputs?

**Comment by gsemet** [comment]
I mostly agree with you:
1. => yes, if you are really motivated to do so,  feel free.
2. => could be done in another task, couldn't it? the mosly awaited one being tab reorg
3. => sure !
4. => yes, that would simplify a lot futur evolutions
5. => I agree with you that terminator (and Tilix) does a pretty good job. But I would like to have something elegant and efficient. Being able at least to split another split (ex: vertical split the screen + the terminal on top being splitted horizontally). Ok to use Paned first but with the option later to switch to a custom made widget (derived from paned?) if the need is. 

For me, the real added value for guake would be to have easy navigation between these tiles. I always found it hard to use on other software (even on iterm2) so if Guake can do less extravagant features but do its set of feature right, I think we will be really happy with it!

**Comment by gsemet** [comment]
yes, let us forget about drag n drop splited terminal for now. My guess is that should be feasible with Paned.

**Comment by aichingm** [comment]
Hey @gsemet,

I have some logistical things to ask. Since this are a lot of changes (I might accidentally break things...) and not only a few bug fixes I would like to open a branch (via PR) on which I can push code via PR so that it can be tested before it is merged into master. Do you have a naming schema you want me to follow? Something like "big-rewrite", "devel", "version-x.x.x" etc...

I just finished phase 1. and fixed #1196. I tested the code as good as I can but since I don't use the task bar very much I don't know all the use cases maybe you know someone who can help testing.

**Comment by gsemet** [comment]
Hello. I have promoted you to “contributor” on Guake, so you should be able to create a branch on the main project. Simply give a meaningful name such as “replace-notebook”.

At work we follow another convention “dev/developer_slug/feature-title”, mine is “dev/gsemet/my-feature”, but for a small project we do not care. 

For me the best is to split your work in several milestones, and merge ASAP each pull request. Having a big big pull request for long time is always tricky to rebase, work with. If we merge the code, at least we have something working even if the job is not finished. Of course the among of “broken” things should be minimal, even if this means to write some “bridge code” that won’t last after the whole work is done.

**Comment by aichingm** [comment]
Thanks for the invite!

I agree with with you completely,  I will make PRs for each of the steps (1-5).
